docs/how_to/users_api.rst

```diff
@@ -0,0 +1,352 @@
+      "default": "Municipal"
+   }
+
+**Field Visibility Options**
+----------------------------
+
+When configuring additional registration fields, there are several visibility and requirement options that can be used:
+
+- **required**: The field is displayed and must be filled out by the user.
+- **optional**: The field is displayed as part of a toggled input field list, and it is not mandatory to fill it out.
+- **hidden**: The field is not displayed to the user.
+- **optional-exposed**: The field is displayed along with the required fields, but filling it out is not mandatory. This option provides more visibility than ``optional`` while still keeping the field optional.
+
+**Testing optional-exposed**
+
+If you want to use and test the ``optional-exposed`` field type, make sure to add it to the configuration. For example, you can set a field to ``optional-exposed`` like this:
+
+.. code-block:: json
+
+   "REGISTRATION_EXTRA_FIELDS": {
+      "org_name": "optional-exposed"
+   }
+
+In this case, the ``org_name`` field will be displayed alongside required fields, but it won't be mandatory for the user to fill out. This can be particularly useful for fields that are not crucial but should be easily visible to users during registration.
+
+**Example Usage**
+
+Here is an example configuration using all the types, including ``optional-exposed``:
+
+.. code-block:: json
+
+   "REGISTRATION_EXTRA_FIELDS": {
+      "confirm_email": "hidden",
+      "level_of_education": "optional",
+      "gender": "optional-exposed",
+      "year_of_birth": "optional",
+      "mailing_address": "optional-exposed",
+      "honor_code": "required"
+   }
+
+In this example:
+
+- ``gender`` and ``mailing_address`` are set to ``optional-exposed``, making them visible alongside required fields but not mandatory.
+- ``honor_code`` is ``required``, ensuring it must be filled.
+- ``level_of_education`` and ``year_of_birth`` are optional and shown in a secondary list.
+- ``confirm_email`` is hidden from the registration form.
+
+**Including the custom field in a POST request:**
+
+.. code-block:: json
+
+   {
+      "username": "johndoe",
+      "email": "johndoe@example.com",
+      "fullname": "John Doe",
+      "password": "p@ssword",
+      "org_name": "Tech Solutions"
+   }
+
+**Including the custom field in a PATCH request:**
+
+.. code-block:: json
+
+   {
+      "email": "johndoe-updated@example.com",
+      "org_name": "New Organization Name"
+   }
+
+By following these steps, the ``org_name`` field will be correctly handled during user creation or update.
+
+
+
+**Update User**
+
+URL: ``/eox-core/api/v1/update-user/``
+
+Method: PATCH
+
+Body:
+
+.. code-block:: json
+
+   {
+      "email": "johndoe-updated@example.com",
+      "password": "updated-p@$$w0rd"
+   }
+
+Example:
+
+.. code-block:: http
+
+   PATCH http://tenant-a.local.edly.io:8000/eox-core/api/v1/update-user/
+
+Response Example:
+
+.. code-block:: json
+
+   200 OK
```

Comment:
I think this endpoint should be on the top, with the other endpoints (GET and POST)

Reply:
Done